### PR TITLE
User list API

### PR DESCRIPTION
**Pull request by Ayush8923**
## Description

- Make user Listing API.
- Make user update API.

### CODE REVIEW

File: app/Http/Controllers/Api/UserController.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+use Illuminate\Support\Facades\Hash;
+
+class UserController extends Controller
+{
+    public function index(): Response
+    {
+        return response(User::paginate(100));
+    }
+
+    public function update(Request $request, User $user)
```
Comment (rathorevaibhav) [inline app/Http/Controllers/Api/UserController.php:18]: Form Request validation is missing

File: routes/api.php [hunk 2]
```diff
@@ -30,6 +31,7 @@
     Route::resource('sender-identity', SenderIdentityController::class)->except(['create', 'edit']);
     Route::resource('campaign', CampaignController::class)->except(['create', 'edit']);
     Route::resource('roles', RoleController::class)->except(['create', 'edit']);
+    Route::resource('users', UserController::class);
```
Comment (rathorevaibhav) [inline routes/api.php:34]: we have singular names in the route. Please update.

Also, the create and edit needs to be put inside `except()`

File: app/Http/Controllers/Api/UserController.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    {
+        $updatedPassword = Hash::make($request->password);
+        $user->update([
+            'name' => $request->name,
+            'email' => $request->email,
+            'password' => $updatedPassword,
+        ]);
+
+        return response()->json(['status' => 'User updated successfully']);
```
Comment (rathorevaibhav) [inline app/Http/Controllers/Api/UserController.php:27]: incorrect resposne for this API

**Review by rathorevaibhav — CHANGES_REQUESTED** [review]

**Comment by Ayush8923** [comment]
Merged this branch into the base branch. Because I need `UserController` in the base branch according to reviewers' feedbacks. That's why I merged this PR into the base branhc.

**Pull request merged**